lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -116,12 +138,19 @@ def with_message(message)
           self
         end
 
+        def allow_nil(allow_nil = true)
+          @allow_nil = allow_nil
+          self
+        end
+
         def matches?(subject)
           super(subject)
           @expected_message ||= :blank
 
           if secure_password_being_validated?
             disallows_and_double_checks_value_of!(blank_value, @expected_message)
+          elsif @allow_nil
+            allows_value_of(nil)
```

Comment:
Yup, looks like I was mistaken about how the `allow_nil` worked here, damn.  I thought it was allowing the associated instance to be `nil`, but ONLY IF the `belongs_to_id` was `nil`, that is, if the instance was not found, it became invalid.

Looks like I can close this PR.  Sorry for your time.